agent/src/beerocks/slave/ap_manager_thread.cpp

```diff
@@ -708,6 +737,51 @@ bool ap_manager_thread::handle_cmdu(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_
         LOG(DEBUG) << "CLIENT_DISALLOW: mac = " << sta_mac << ", bssid = " << bssid;
 
         ap_wlan_hal->sta_deny(sta_mac, bssid);
+
+        // Check if validity period is set then add it to the "disallowed client timeouts" list
+        // This list will be polled in after_select()
+        // When validity period is timed out CLIENT_ALLOW message will be sent.
+        if (request->validity_period()) {
+
+            disallowed_client_params_t disallowed_client_params;
+            disallowed_client_params.mac   = request->mac();
+            disallowed_client_params.bssid = request->bssid();
+
+            // check if STA is already disallowed from this bssid
+            auto disallowed_client_timeout = std::find_if(m_disallowed_client_timeouts.begin(), m_disallowed_client_timeouts.end(),
+                        [&](const std::pair<std::chrono::steady_clock::time_point, disallowed_client_params_t> &element) {
+                            return ((element.second.mac == disallowed_client_params.mac) && (element.second.bssid == disallowed_client_params.bssid));
```

Comment:
Unless I'm mistaken, the default operator== of disallowed_client_params_t should do exactly this. In other words:
```suggestion
                            return ((element.second == disallowed_client_params));
```

However, my other comment about adding timeout to the struct will make this no longer true, unless you add an explicit operator== that ignores the timeout.